snapcraft/storeapi/__init__.py

```diff
@@ -432,6 +444,51 @@ def snap_release(self, snap_name, revision, channels):
 
         return response_json
 
+    def push_validation(self, snap_id, assertion):
+        data = {
+            'assertion': assertion.decode('utf-8'),
+        }
+        auth = _macaroon_auth(self.conf)
+        response = self.put(
+            'snaps/{}/validations'.format(snap_id), data=json.dumps(data),
+            headers={'Authorization': auth,
+                     'Content-Type': 'application/json',
+                     'Accept': 'application/json'})
+        if not response.ok:
+            raise errors.StoreValidationError(snap_id, response)
+        try:
+            response_json = response.json()
+        except JSONDecodeError:
+            message = ('Invalid response from the server when pushing '
+                       'validations: {} {}').format(
+                           response.status_code, response)
+            logger.debug(message)
```

Comment:
this is hardly a debug message, it's an error. What about making sure the raised exception has enough information to guide the user and that's it.
If our servers are not returning proper JSON and we don't know already ... it's more like our problem.